mobile/redux/features/signinSlice.ts

```diff
@@ -48,6 +54,51 @@ export const signIn = createAsyncThunk<boolean, SignInParam, ThunkExtra>("signin
   return true;
 });
 
+export const changeMasterPassword = createAsyncThunk<string, ChangeMasterParam, ThunkExtra>("user/changeMasterPass", async (param, config) => {
+  const { newPassword } = param;
+  const { masterPassword } = (config.getState() as RootState).signIn
+
+  const gnonative = config.extra.gnonative as GnoNativeApi;
+
+  if (!newPassword) {
+    throw new Error("newPassword is required.")
+  }
+
+  if (!masterPassword) {
+    throw new Error("Master password not found.");
+  }
+
+  try {
+    const response = await gnonative.listKeyInfo();
+
+    if (response.length === 0) {
+      throw new Error("No accounts found.");
+    }
+
+    for (const account of response) {
+      console.log("change password for account", account);
+      await gnonative.selectAccount(account.name);
+      await gnonative.setPassword(masterPassword);
+      await gnonative.updatePassword(newPassword);
```

Comment:
Suppose there are 2 keys in the keybase. If the loop updates key 1, but `updatePassword` fails on key 2, then the loop will quit and leave the keybase in an inconsistent state. (Different keys with different passwords.) Do you think this should be some logic to handle this case?

Reply:
Absolutely.